### PR TITLE
ref(tsc): Convert AccountSecurityDetails to FC

### PR DESCRIPTION
**Pull request by ArthurKnaus**
Convert `<AccountSecurityDetails />` to FC.

- part of https://github.com/getsentry/frontend-tsc/issues/2

### CODE REVIEW

File: static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx [hunk 1]
```diff
@@ -43,160 +49,165 @@ function AuthenticatorDate({label, date}: AuthenticatorDateProps) {
...
-        )}
-
-        <RecoveryCodes
-          onRegenerateBackupCodes={onRegenerateBackupCodes}
-          isEnrolled={authenticator.isEnrolled}
-          codes={authenticator.codes}
-        />
-      </Fragment>
-    );
+  if (!authenticator || isError) {
```
Comment (priscilawebdev) [inline static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:139]: shouldn't we check only for `isError` here?

File: static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx [hunk 2]
```diff
@@ -43,160 +49,165 @@ function AuthenticatorDate({label, date}: AuthenticatorDateProps) {
...
+    return <LoadingError onRetry={refetch} />;
   }
+
+  return (
+    <SentryDocumentTitle title={PAGE_TITLE}>
```
Comment (priscilawebdev) [inline static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:144]: hm why not inline the page title?

File: static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:139]: and if an error happens in the mutations?

Comment (ArthurKnaus) [reply static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:139]: The previous code handled the response as if the endpoint could return `null`.
As far as I see from the python code that should not be possible, will adjust the types.

**Review by priscilawebdev — APPROVED** [review]
🙏

**Pull request merged**